csrc/tensor_view.cpp

```diff
@@ -1198,17 +1198,43 @@ void TensorView::clearReductionIterDomains() {
       getLeafDomain() == getRootDomain(),
       "should not call clearReductionIterDomains on already transformed TensorDomains");
 
+  std::vector<IterDomain*> root = getRootDomain();
+  std::vector<IterDomain*> alloc = getMaybeAllocationDomain();
+
+  NVF_ERROR(
+      std::is_permutation(root.begin(), root.end(), alloc.begin(), alloc.end()),
+      "should not call clearReductionIterDomains on transformed allocation domain");
+
   std::vector<IterDomain*> new_root;
+  std::vector<IterDomain*> new_alloc;
   std::vector<std::optional<bool>> new_contig;
   for (const auto i : c10::irange(getRootDomain().size())) {
     auto root_i = getRootDomain().at(i);
     if (!root_i->isReduction()) {
       new_root.push_back(root_i);
+    }
+    // contig flag is specified for on allocation domain
+    auto alloc_i = getAllocationDomain().at(i);
```

Comment:
Is allocation domain always specified?

Reply:
Good catch. This one should use `alloc.at(i)` instead.